packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -307,8 +307,8 @@ class DefaultTextEditingShortcuts extends Shortcuts {
     const SingleActivator(LogicalKeyboardKey.arrowUp, meta: true): const ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: true),
     const SingleActivator(LogicalKeyboardKey.arrowDown, meta: true): const ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: true),
 
-    const SingleActivator(LogicalKeyboardKey.arrowLeft, shift: true, meta: true): const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false),
-    const SingleActivator(LogicalKeyboardKey.arrowRight, shift: true, meta: true): const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false),
+    const SingleActivator(LogicalKeyboardKey.arrowLeft, shift: true, meta: true): const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false, expand: true),
```

Comment:
@LongCatIsLooong What do you think of adding this expand parameter here, or would it be better to do something like creating an all new intent?

Reply:
Seems reasonable, but I would slightly prefer having a different intent since `collapseSelection` and `expand` can't both be true and currently it relies on a run time assert.

Reply:
I've created a new Intent as a subclass of DirectionalTextEditingIntent and I think it worked out cleaner 👍 .